### PR TITLE
ref(controller): use the env config instead of directly getting from environment

### LINKED ISSUE #338: refactor the controller utils to the controller env variables
# Summary

currently controller utils(https://github.com/deis/builder/blob/master/pkg/controller/utils.go#L30)  directly gets the values of the environment values instead we should get it through the global configuration file we have(https://github.com/deis/builder/blob/master/pkg/conf/config.go#L29)

## Everything Else
# Related Issues
ref https://github.com/deis/builder/pull/335#discussion_r63078027

### PR DESCRIPTION
**Pull request by kmala**
fixes #338

### CODE REVIEW

**Comment by bacongobbler** [comment]
can you try to write unit tests for this if possible? Trying to increase coverage in #437 so it'd be helpful to have this all covered before merging.

**Comment by kmala** [comment]
@bacongobbler i had added test.Can you review it again.

File: pkg/conf/config.go [hunk 1]
```diff
@@ -36,10 +36,10 @@ func EnvConfig(appName string, conf interface{}) error {
 }
 
 // GetBuilderKey returns the key to be used as token to interact with deis-controller
-func GetBuilderKey() (string, error) {
-	builderKeyBytes, err := ioutil.ReadFile(builderKeyLocation)
+func GetBuilderKey(builderKeyPath string) (string, error) {
```
Comment (bacongobbler) [inline pkg/conf/config.go]: Would it be easier to change BuilderKeyLocation to type `var` so tests can change the path before running `GetBuilderKey()`? Wondering why we'd define a constant `BuilderKeyLocation` variable, yet not use it for `GetBuilderKey()`.

Comment (bacongobbler) [inline pkg/conf/config.go]: FYI I'm running into this same problem in #437 as well.

Comment (kmala) [inline pkg/conf/config.go]: Just to be on the safeside that it we don't change it by mistake in other part of the code.I am sure that doesn't happen so i am ok with changing it to `var`

**Comment by bacongobbler** [comment]
LGTM even without the change. This is partially blocking #437 as it touches on some of the code I'm directly testing in build.go (in a good way) :+1:

**Pull request merged**